### PR TITLE
fix TypeError: Cannot read property 'app_id' of undefined + add assertValidAgoraConfig

### PR DESCRIPTION
**Pull request by 0xdevalias**
fixes `TypeError: Cannot read property 'app_id' of undefined` that happens when deploying functions to an environment that doesn't have the agora config keys (introduced in #1673) in it's function config:

```
Error: Error occurred while parsing your function triggers.

TypeError: Cannot read property 'app_id' of undefined
    at Object.<anonymous> (/home/circleci/project/functions/src/utils/agora.js:7:28)
    at Module._compile (internal/modules/cjs/loader.js:1147:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
    at Module.load (internal/modules/cjs/loader.js:996:32)
    at Function.Module._load (internal/modules/cjs/loader.js:896:14)
    at Module.require (internal/modules/cjs/loader.js:1036:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/circleci/project/functions/video.js:4:51)
    at Module._compile (internal/modules/cjs/loader.js:1147:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
```

eg. https://app.circleci.com/pipelines/github/sparkletown/sparkle/7911/workflows/532f21e2-65f5-48fc-8307-f6d4530fa80d/jobs/12245

---

partially fixes sparkletown/internal-sparkle-issues#748

partially fixes sparkletown/internal-sparkle-issues#859

### CODE REVIEW

File: functions/video.js [hunk 1]
```diff
@@ -32,6 +36,19 @@ exports.getAgoraToken = functions.https.onCall((data, context) => {
     );
   }
 
+  try {
+    assertValidAgoraConfig();
+  } catch (err) {
```
Comment (sunny-viktoryia) [inline functions/video.js:41]: looks like you want to use `err` in the logger few lines below. Or is there another `error` variable?
```suggestion
  } catch (error) {
```

**Review by sunny-viktoryia — APPROVED** [review]
Approving, but have a couple of questions. 🙃 
Just want to confirm if the changes are correct.

File: functions/src/utils/agora.js [hunk 2]
```diff
@@ -2,21 +2,37 @@ const functions = require("firebase-functions");
...
-const appCertificate = AGORA_CONFIG.app_certificate;
+const AGORA_CONFIG = functions.config().agora || {};
+const AGORA_APP_ID = AGORA_CONFIG.app_id;
+const AGORA_APP_CERTIFICATE = AGORA_CONFIG.app_certificate;
 
 // @debt should we move this into AGORA_CONFIG, or maybe venue config in firestore or similar?
 const expirationTimeInSeconds = 3600;
 
+const assertValidAgoraConfig = () => {
+  if (typeof AGORA_APP_ID !== "string" && AGORA_APP_ID.length <= 0)
```
[the remark this answers is not in the document]
Comment (0xdevalias) [reply functions/src/utils/agora.js]: Nope, you're correct. And the ones I copied this from were wrong too.

All fixed in d04d7e172f7098a47695874b6744ebe0818a410d

File: functions/src/utils/agora.js [hunk 3]
```diff
@@ -2,21 +2,37 @@ const functions = require("firebase-functions");
...
+    throw new Error(
+      "required configuration missing, AGORA_APP_ID must be a string"
+    );
+
+  if (
+    typeof AGORA_APP_CERTIFICATE !== "string" &&
```
[the remark this answers is not in the document]
Comment (0xdevalias) [reply functions/src/utils/agora.js]: Nope, you're correct. And the ones I copied this from were wrong too.

All fixed in d04d7e172f7098a47695874b6744ebe0818a410d

File: functions/video.js (hunk 1 above)
Comment (0xdevalias) [reply functions/video.js]: Nice catch!

Fixed in c0a937a6a2acba7b5cc929066d14eaabfe9d2249

**Review by sunny-viktoryia — APPROVED** [review]
Looks good to me now. 🙃 Thank you!

**Pull request merged**